subsys/bluetooth/audio/has.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+#include <device.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/audio/audio.h>
+#include <bluetooth/audio/capabilities.h>
+
+#include "has_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HAS)
+#define LOG_MODULE_NAME bt_has
+#include "common/log.h"
+
+static struct bt_has has;
+
+static ssize_t read_features(struct bt_conn *conn, const struct bt_gatt_attr *attr, void *buf,
+			     uint16_t len, uint16_t offset)
+{
+	BT_DBG("conn %p attr %p offset %d", conn, attr, offset);
+
+	if (offset > sizeof(has.features)) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
+	}
+
+	return bt_gatt_attr_read(conn, attr, buf, len, offset, &has.features,
+				 sizeof(has.features));
+}
+
+/* Hearing Access Service GATT Attributes */
+BT_GATT_SERVICE_DEFINE(has_svc,
+	BT_GATT_PRIMARY_SERVICE(BT_UUID_HAS),
+	BT_GATT_CHARACTERISTIC(BT_UUID_HAS_HEARING_AID_FEATURES, BT_GATT_CHRC_READ,
+			       BT_GATT_PERM_READ_ENCRYPT, read_features, NULL, NULL),
+);
+
+static int has_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* Initialize the supported features characteristic value */
+	has.features = CONFIG_BT_HAS_HEARING_AID_TYPE;
+
+	if (IS_ENABLED(CONFIG_BT_HAS_HEARING_AID_BANDED)) {
+		/* HAP_d1.0r00; 3.7 BAP Unicast Server role requirements
+		 * A Banded Hearing Aid in the HA role shall set the Front Left and the Front
+		 * Right bits to a value of 0b1 in the Sink Audio Locations characteristic value.
+		 */
+		bt_audio_capability_set_location(BT_AUDIO_SINK, BT_AUDIO_LOCATION_FRONT_LEFT |
+								BT_AUDIO_LOCATION_FRONT_RIGHT);
```

Comment:
Should we also set the appearance? 

e.g. via `bt_set_appearance` or `CONFIG_BT_DEVICE_APPEARANCE`, or just leave that to the application?

Reply:
I guess this should be leaved to the application.
The specification mentions Appearance AD Type, so that this has to be done by app then the advertising is started.
Neither bt_set_appearance nor CONFIG_BT_DEVICE_APPEARANCE will set the appearance field in ADV

Reply:
Does the device advertise the BT_AUDIO_LOCATION value? If so, then this should also be set by the application, else this is fine :)

Reply:
Neither HAP not HAS specification mention about it.

Reply:
If yes, `bt_audio_capability_get_location` is needed then.

Reply:
By the way, looks like ASCS says ASCS UUID should be included in ext advertising data. Is that handled by a stack or left to the application?